src/client/mod.rs

```diff
@@ -307,8 +320,15 @@ impl Client {
     pub async fn watch(
         &self,
         pipeline: impl IntoIterator<Item = Document>,
-        options: Option<ChangeStreamOptions>,
+        options: impl Into<Option<ChangeStreamOptions>>,
     ) -> Result<ChangeStream> {
-        todo!();
+        let options = options.into();
+        resolve_options!(self, options.clone(), [read_concern, selection_criteria]);
```

Comment:
we don't want to clone here since we want to mutate the options we already have. This applies to other invocations of `resolve_options` too.

Reply:
fixed